### PR TITLE
Fix #5193 Backend - Need support to Filter users rest api with isAdmin field

### LINKED ISSUE #5193: Backend - Need support to Filter users rest api with isAdmin field
**Is your feature request related to a problem? Please describe.**
Currently users api do not support passing of any fields to filter only admin users

**Describe the solution you'd like**
users api to support filtering of admin users like table api supports filtering with database

**Describe alternatives you've considered**
NA

**Additional context**
NA

### LINKED ISSUE #5349: UI: Teams and Users page takes a long time to load
It still has a 100000 limit call to users API and takes a long time to load

### PR DESCRIPTION
**Pull request by vivekratnavel**
Closes #5193 
Closes #5349 

### Describe your changes :
See #5193 

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/TeamsAndUsersPage/TeamsAndUsersPage.component.tsx [hunk 1]
```diff
@@ -103,23 +102,41 @@ const TeamsAndUsersPage = () => {
...
 
   /**
    * Make API call to fetch all the users
    */
   const fetchUsers = (
-    limit = API_RES_MAX_SIZE,
-    paging = {} as Record<string, string>
+    limit = PAGE_SIZE_BASE,
+    paging = {} as Record<string, string>,
+    isAdmin = undefined as boolean | undefined
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/TeamsAndUsersPage/TeamsAndUsersPage.component.tsx:132]: Do we need this?
We can directly pass the `undefined`.

File: openmetadata-ui/src/main/resources/ui/src/pages/TeamsAndUsersPage/TeamsAndUsersPage.component.tsx [hunk 2]
```diff
@@ -166,50 +192,29 @@ const TeamsAndUsersPage = () => {
...
-  };
-
   const setAllTabList = (type: string, cursorValue?: string | number) => {
     setIsUsersLoading(true);
 
     if (type) {
-      if (type === UserType.ADMINS) {
-        const dAdmins = getAdmins();
-        setSelectedUserList(dAdmins);
+      const isAdmin = type === UserType.ADMINS ? true : undefined;
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/TeamsAndUsersPage/TeamsAndUsersPage.component.tsx:199]:
```suggestion
      const isAdmin = type === UserType.ADMINS || undefined;
```

File: openmetadata-ui/src/main/resources/ui/src/pages/TeamsAndUsersPage/TeamsAndUsersPage.component.tsx (hunk 1 above)
Comment (vivekratnavel) [reply openmetadata-ui/src/main/resources/ui/src/pages/TeamsAndUsersPage/TeamsAndUsersPage.component.tsx:132]: Yes, we need it to keep typescript happy :)
Screen Shot 2022-07-18 at 10 53 58 PM

Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/TeamsAndUsersPage/TeamsAndUsersPage.component.tsx:132]: i think we can use `isAdmin = false`

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**